src/Microsoft.Management.Infrastructure.CimCmdlets/CimAsyncOperation.cs

```diff
@@ -107,13 +104,9 @@ protected void OperationDeletedHandler(object cimSession, OperationEventArgs act
         {
             DebugHelper.WriteLogEx();
 
-            lock (this.a_lock)
+            if (Interlocked.Decrement(ref this.operationCount) == 0)
             {
-                this.operationCount--;
-                if (this.operationCount == 0)
-                {
-                    this.moreActionEvent.Set();
-                }
+                this.moreActionEvent.Set();
```

Comment:
This was under lock before.

Reply:
I see that other `Set`, `Wait`, `Reset` calls on the `moreActionEvent` object are not protected by a lock, therefore I assume the original intent was to protect only the `operationCount` field.